apps/frontend/src/app/5_pages/AavePage/components/LendPositionsList/components/WithdrawModal/WithdrawForm.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { FC, useCallback, useMemo, useState } from 'react';
+
+import { t } from 'i18next';
+
+import {
+  AmountInput,
+  Button,
+  ErrorBadge,
+  ErrorLevel,
+  Select,
+  SimpleTable,
+  SimpleTableRow,
+  Tabs,
+  TabType,
+} from '@sovryn/ui';
+import { Decimal } from '@sovryn/utils';
+
+import { AmountRenderer } from '../../../../../../2_molecules/AmountRenderer/AmountRenderer';
+import { AssetRenderer } from '../../../../../../2_molecules/AssetRenderer/AssetRenderer';
+import { useDecimalAmountInput } from '../../../../../../../hooks/useDecimalAmountInput';
+import { translations } from '../../../../../../../locales/i18n';
+
+const pageTranslations = translations.aavePage;
+
+type WithdrawFormProps = {
+  onSuccess: () => unknown;
+};
+
+export const WithdrawForm: FC<WithdrawFormProps> = () => {
+  const withdrawableAssets = useMemo(() => ['BTC', 'SOV'], []); // TODO: this is mocked data. Replace with proper hook
+  const [maximumWithdrawAmount] = useState<Decimal>(Decimal.from(10)); // TODO: this is mocked data. Replace with proper hook
+  const [withdrawAsset, setWithdrawAsset] = useState<string>(
+    withdrawableAssets[0],
+  );
+  const [withdrawAmount, setWithdrawAmount, withdrawSize] =
+    useDecimalAmountInput('');
+
+  const onWithdrawAssetChange = useCallback(v => {
+    setWithdrawAsset(v);
+  }, []);
+
+  const withdrawableAssetsOptions = useMemo(
+    () =>
+      withdrawableAssets.map(token => ({
+        value: token,
+        label: (
+          <AssetRenderer
+            showAssetLogo
+            asset={token}
+            assetClassName="font-medium"
+          />
+        ),
+      })),
+    [withdrawableAssets],
+  );
+
+  const isValidWithdrawAmount = useMemo(
+    () => (withdrawSize.gt(0) ? withdrawSize.lte(maximumWithdrawAmount) : true),
+    [withdrawSize, maximumWithdrawAmount],
+  );
+
+  const remainingSupply = useMemo(
+    () => maximumWithdrawAmount.sub(withdrawSize),
+    [withdrawSize, maximumWithdrawAmount],
+  );
```

Comment:
same to useCallback. Check if it is necessary to avoid others renders, otherwise recalculate without worry.

Reply:
Well agree, but seems like they use it pretty intensively
Example of same operation -> https://github.com/DistributedCollective/sovryn-dapp/blob/af5d8b320f31a6afe1928229de5fbe2c27004c3b/apps/frontend/src/app/5_pages/BorrowPage/components/NewLoanForm/NewLoanForm.tsx#L184 

Maybe because it's Decimal instances and there will be a lot more going on so maybe we keep this one even just for consistency...

Reply:
Great! So forget about it!
Follow their rules

Reply:
Perfect! I think is a valid point anyways, thanks for bringing it up!!